nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -27,118 +24,47 @@ namespace nav2_bt_navigator
-  for (;; ) {
-    // Check to see if this task (navigation) has been canceled. If so, cancel any child
-    // tasks and then cancel this task
-    if (cancelRequested()) {
-      RCLCPP_INFO(get_logger(), "BtNavigator: Task has been canceled.");
-      plannerTaskClient_->cancel();
-      setCanceled();
-      return TaskStatus::CANCELED;
-    }
-
-    // Check if the planning task has completed
-    TaskStatus status = plannerTaskClient_->waitForResult(path, 100ms);
-
-    switch (status) {
-      case TaskStatus::SUCCEEDED:
-        RCLCPP_INFO(get_logger(), "BtNavigator: Planning task completed.");
-        goto here;
-
-      case TaskStatus::FAILED:
-        return TaskStatus::FAILED;
-
-      case TaskStatus::RUNNING:
-        RCLCPP_DEBUG(get_logger(), "BtNavigator: Planning task still running");
-        break;
-
-      default:
-        RCLCPP_ERROR(get_logger(), "BtNavigator: Invalid status value.");
-        throw std::logic_error("BtNavigator::execute: invalid status value");
-    }
-  }
-
-here:
-  RCLCPP_INFO(get_logger(),
-    "BtNavigator: Sending the path to the controller to execute.");
-
-  controllerTaskClient_->sendCommand(path);
-
-  // Loop until the control task completes
-  for (;; ) {
-    // Check to see if this task (navigation) has been canceled. If so, cancel any child
-    // tasks and then cancel this task
-    if (cancelRequested()) {
-      RCLCPP_INFO(get_logger(), "BtNavigator: Task has been canceled.");
-      controllerTaskClient_->cancel();
-      setCanceled();
-      return TaskStatus::CANCELED;
-    }
-
-    // Check if the control task has completed
-    auto controlResult = std::make_shared<nav2_tasks::FollowPathResult>();
-    TaskStatus status = controllerTaskClient_->waitForResult(controlResult, 10ms);
-
-    switch (status) {
-      case TaskStatus::SUCCEEDED:
-        {
-          RCLCPP_INFO(get_logger(), "BtNavigator: Control task completed.");
-          nav2_tasks::NavigateToPoseResult navigationResult;
-          setResult(navigationResult);
-
-          return TaskStatus::SUCCEEDED;
-        }
-
-      case TaskStatus::FAILED:
-        return TaskStatus::FAILED;
-
-      case TaskStatus::RUNNING:
-        RCLCPP_INFO(get_logger(), "BtNavigator: Control task still running.");
-        break;
-
-      default:
-        RCLCPP_ERROR(get_logger(), "BtNavigator: Invalid status value.");
-        throw std::logic_error("BtNavigator::execute: invalid status value");
-    }
-  }
+  // Get a reference for convenience
+  geometry_msgs::msg::Pose & p = command->pose;
+
+  // Compose the args for the ComputePathToPose action
+  std::stringstream args;
+  args << "position=\"" <<
+    p.position.x << ";" << p.position.y << ";" << p.position.z << "\"" <<
+    " orientation=\"" <<
+    p.orientation.x << ";" << p.orientation.y << ";" <<
+    p.orientation.z << ";" << p.orientation.w << "\"";
+
+  // Put it all together, trying to make the XML somewhat readable here
+  std::stringstream ss;
```

Comment:
Can you change the variable name to 'navigation_mission" or something similar?

Reply:
Fixed.